libs/src/api/Track.ts

```diff
@@ -411,21 +419,36 @@ export class Track extends Base {
       phase = phases.ADDING_TRACK
 
       // Write metadata to chain
-      const multihashDecoded = Utils.decodeMultihash(metadataMultihash)
-      const { txReceipt, trackId } =
-        await this.contracts.TrackFactoryClient.addTrack(
+      let txReceipt: TransactionReceipt
+      let trackId: number
+      if (useEntityManager) {
+        trackId = Track.generateTrackId()
+        const response = await this.contracts.EntityManagerClient!.manageEntity(
+          ownerId,
+          EntityManagerClient.EntityType.TRACK,
+          trackId,
+          EntityManagerClient.Action.CREATE,
+          metadataMultihash
+        )
+        txReceipt = response.txReceipt
+      } else {
+        const multihashDecoded = Utils.decodeMultihash(metadataMultihash)
+        const response = await this.contracts.TrackFactoryClient.addTrack(
           ownerId,
           multihashDecoded.digest,
           multihashDecoded.hashFn,
           multihashDecoded.size
         )
+        txReceipt = response.txReceipt
+        trackId = response.trackId
+      }
 
       phase = phases.ASSOCIATING_TRACK
       // Associate the track id with the file metadata and block number
       await this.creatorNode.associateTrack(
-        trackId,
+        trackId!,
```

Comment:
is there a clean way to avoid using `!`, like having if checks and error logs if they aren't there?